packetbeat/sniffer/afpacket_linux.go

```diff
@@ -69,4 +87,42 @@ func (h *afpacketHandle) LinkType() layers.LinkType {
 
 func (h *afpacketHandle) Close() {
 	h.TPacket.Close()
+	if err := setPromiscMode(h.device, h.promicsPreviousState); err != nil {
+		logp.Err("Failed to set promiscuous mode for device '%s': %v", h.device, err)
+	}
+}
+
+func isPromiscEnabled(device string) (bool, error) {
+	if device == "any" {
+		return false, nil
+	}
+
+	s, e := syscall.Socket(syscall.AF_INET, syscall.SOCK_DGRAM, 0)
+	if e != nil {
+		return false, e
+	}
+
+	defer syscall.Close(s)
+
+	var ifl struct {
+		name  [syscall.IFNAMSIZ]byte
+		flags uint16
+	}
+
+	copy(ifl.name[:], []byte(device))
+	_, _, ep := syscall.Syscall(syscall.SYS_IOCTL, uintptr(s), syscall.SIOCGIFFLAGS, uintptr(unsafe.Pointer(&ifl)))
+	if ep != 0 {
+		return false, fmt.Errorf("Syscall SIOCGIFFLAGS exited with %v", ep)
+	}
+
+	return ifl.flags&uint16(syscall.IFF_PROMISC) != 0, nil
+}
+
+func setPromiscMode(device string, enabled bool) error {
+	if device == "any" {
+		logp.Warn("Cannot set promiscuous mode to device 'any'")
+		return nil
+	}
+
+	return syscall.SetLsfPromisc(device, enabled)
```

Comment:
This function says its deprecated. Have you checked out what it recommends to use?

Reply:
for now `x/net` where I did not find anything useful. so, for now, I went with this with a possible rewrite to direct syscalls

Reply:
Let's at least add a code comment here explaining the issue + follow up issue in repo.